streams/src/main/java/org/apache/kafka/streams/kstream/internals/KGroupedStreamImpl.java

```diff
@@ -43,29 +43,26 @@
 
     private final Serde<K> keySerde;
     private final Serde<V> valSerde;
-    private final boolean repartitionRequired;
     private final GroupedStreamAggregateBuilder<K, V> aggregateBuilder;
 
     KGroupedStreamImpl(final InternalStreamsBuilder builder,
                        final String name,
                        final Set<String> sourceNodes,
-                       final Serde<K> keySerde,
-                       final Serde<V> valSerde,
+                       final GroupedInternal<K, V> groupedInternal,
                        final boolean repartitionRequired,
                        final StreamsGraphNode streamsGraphNode) {
         super(builder, name, sourceNodes, streamsGraphNode);
         this.aggregateBuilder = new GroupedStreamAggregateBuilder<>(
             builder,
-            keySerde,
-            valSerde,
+            groupedInternal.keySerde(),
+            groupedInternal.valueSerde(),
             repartitionRequired,
             sourceNodes,
             name,
-            streamsGraphNode
-        );
-        this.keySerde = keySerde;
-        this.valSerde = valSerde;
-        this.repartitionRequired = repartitionRequired;
+            streamsGraphNode,
+                groupedInternal.name());
```

Comment:
nit: alignment

Reply:
nit: Since the `GroupedStreamAggregateBuilder` needs to know everything that's in `GroupedInternal`, maybe we can just pass the whole config object in to cut down on the param list?

Reply:
I had the same thought, I think this just slipped.